parsers/US_NEISO.py

```diff
@@ -97,50 +108,53 @@ def production_data_processer(raw_data, logger: Logger) -> list:
             # passing None to arrow.get() will return current time
             counter += 1
             logger.warning(
-                "Skipping US-NEISO datapoint missing timestamp.",
-                extra={"key": "US-NEISO"},
+                f"Skipping {zone_key} datapoint missing timestamp.",
+                extra={"zone_key": zone_key},
             )
             continue
 
         # neiso storage flow signs are opposite to EM
         battery_storage = -1 * datapoint.pop("Other", 0.0)
+        storage_mix = StorageMix(battery=battery_storage, hydro=None)
 
-        production = defaultdict(lambda: 0.0)
+        production = {mode: None for mode in generation_mapping.values()}
         for k, v in datapoint.items():
-            # Need to avoid duplicate keys overwriting.
-            production[generation_mapping[k]] += v
-
-        # move small negative values to 0
-        for k, v in production.items():
-            if -5 < v < 0:
-                production[k] = 0
-
-        clean_data.append((dt, dict(production), battery_storage))
+            if production[generation_mapping[k]] is None:
+                production[generation_mapping[k]] = v
+            else:
+                production[generation_mapping[k]] += v
+        production_mix = ProductionMix(**production)
```

Comment:
In order to unify this across parsers and reduce code duplication we added the `.add_value()` function to ProductionMix (and StorageMix) a few weeks ago. I would appriciate if we used that instead as it has some additional error correction logging.

Reply:
I'll let you check that I used it as is intended :)